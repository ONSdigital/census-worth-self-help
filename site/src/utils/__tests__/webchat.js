import WebChat from "../webchat"
import React from "react"
import renderer from "react-test-renderer"
import { render } from "react-testing-library"

describe("div rendering", () => {
  beforeEach(() => {
    process.env.GATSBY_CHAT_BUTTON_CONTAINER_ID = "chat-button-container"
  });

  it("renders normally when online", () => {
    const tree = renderer.create(<WebChat />).toJSON()
    expect(tree).toMatchSnapshot()

    const { getByTestId } = render(<WebChat />)
    expect(getByTestId("webchat-link")).not.toHaveStyle("display:none")
  })

  it("is not displayed when offline", () => {
  	Object.defineProperty(navigator, "onLine", {
      configurable: true,
      value: false
    });

    const { getByTestId } = render(<WebChat />)
    expect(getByTestId("webchat-link")).toHaveStyle("display:none")
  })
})