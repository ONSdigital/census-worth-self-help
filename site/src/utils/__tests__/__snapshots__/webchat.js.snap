// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div rendering renders normally when online 1`] = `
<div
  className="webchat-link"
  data-testid="webchat-link"
  style={
    Object {
      "display": "",
    }
  }
>
  <div
    id="chat-button-container"
  />
</div>
`;
